lib/livebook_web/live/settings_live.ex

```diff
@@ -357,8 +356,21 @@ defmodule LivebookWeb.SettingsLive do
     handle_event("set_autosave_path", %{}, socket)
   end
 
-  def handle_info({:env_vars_changed, env_vars}, socket) do
-    {:noreply, assign(socket, env_vars: env_vars, env_var: nil)}
+  def handle_info({:env_var_set, env_var}, socket) do
+    idx = Enum.find_index(socket.assigns.env_vars, &(&1.key == env_var.key))
+
+    env_vars =
+      if idx,
+        do: List.replace_at(socket.assigns.env_vars, idx, env_var),
+        else: [env_var | socket.assigns.env_vars]
+
+    {:noreply, assign(socket, env_vars: Enum.sort(env_vars), env_var: nil)}
+  end
+
+  def handle_info({:env_var_unset, env_var}, socket) do
+    env_vars = Enum.reject(socket.assigns.env_vars, &(&1.key == env_var.key))
+
+    {:noreply, assign(socket, env_vars: Enum.sort(env_vars), env_var: nil)}
```

Comment:
We should sort them on init, right? And then we don't need to resort here :)

Reply:
Yeah, and we already sort on add/update, so it's okay to no resort here